group-1/homeworks/igor.pavloski/task-1/script.js

```diff
@@ -0,0 +1,77 @@
+//Task 1:
+
+let button = document.getElementById("roll");
+let result = document.getElementById("result");
+let rollRandomNumber = (max) => Math.random() * (max + 1) | 0;
+
+function loadJSON(path, success, error) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            if (xhr.status === 200) {
+                if (success)
+                    success(JSON.parse(xhr.responseText));
+            } else {
+                if (error)
+                    error(xhr);
+            }
+        }
+    };
+    xhr.open("GET", path, true);
+    xhr.send();
+}
+
+
+
+button.addEventListener("click", function () {
+
+    loadJSON('filelist.json',
+        function (fileList) {
+            let randomFile = fileList[rollRandomNumber(19)];
+
+            loadJSON(randomFile,
+                function (randomFile) {
+                    let sum = 0;
+
+                    for (i = 0; i < randomFile.data.length; i++) {
```

Comment:
Note that the operation will be constant throughout the processing, so if it is "sine" it will be "sine" for all the elements. And the switch will execute for each and every element.

Try to refactor the code, so that the switch is executed only once, and it determines the function that will be mapped to all the elements.